packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -33,7 +33,12 @@ export type FormFieldConfig<T> = (
     | { type: "date" }
     // TODO | { type: "dateTime" }
     | { type: "staticSelect"; values?: Array<{ value: string; label: string } | string> }
-    | { type: "asyncSelect"; rootQuery: string; labelField?: string }
+    | {
+          type: "asyncSelect";
+          rootQuery: string;
+          labelField?: string;
+          filterField?: { name: string; gqlName?: string };
```

Comment:
@nsams you mentioned infering `gqlVarName` and `gqlVarType` in this comment https://github.com/vivid-planet/comet/pull/2404#discussion_r1712573051

because this config based on this pull-request I refactored this pull-request. (this commit: [Refactor generating asyncSelect-filter](https://github.com/vivid-planet/comet/pull/2372/commits/a42b7a10d4c343006f1967352421fdd16157ad7f) )